apps/src/templates/amazonFutureEngineerEligibilityForm.jsx

```diff
@@ -0,0 +1,210 @@
+              id="firstName"
+              label="First name"
+              type="text"
+              required={true}
+              onChange={this.handleChange}
+            />
+            <FieldGroup
+              id="lastName"
+              label="Last name"
+              type="text"
+              required={true}
+              onChange={this.handleChange}
+            />
+            <div>
+              How can Amazon Future Engineer help you grow computer science at
+              your school?
+            </div>
+            <hr />
+            <SingleCheckbox
+              name="inspirationKit"
+              label="Send my school an Inspiration Kit with posters and stickers to
+              help promote computer science to students and parents."
+              onChange={this.handleChange}
+              value={this.state.inspirationKit}
+            />
+            {this.state.inspirationKit && (
+              <ShippingAddressFormGroup handleChange={this.handleChange} />
+            )}
+            <SingleCheckbox
+              name="csta"
+              label="Send me a free annual Computer Science Teachers Association (CSTA)
+              Plus membership - which includes access to Amazon expert-led
+              webinars and other exclusive content."
+              onChange={this.handleChange}
+              value={this.state.csta}
+            />
+            <SingleCheckbox
+              name="awsEducate"
+              label="Send me a free membership to Amazon Web Services Educate to access
+              free content and cloud computing credits to help my students learn
+              to build in the cloud."
+              onChange={this.handleChange}
+              value={this.state.awsEducate}
+            />
+            <hr />
+            <SingleCheckbox
+              name="consent"
+              label="I give Code.org permission to share my name, email address, and
+              school name, address, and ID with Amazon.com (required to
+              participate). Use of your personal information is subject to
+              Amazon’s Privacy Policy."
+              onChange={this.handleChange}
+              value={this.state.consent}
+              required={true}
+            />
+            <div>
+              By clicking Continue, you will receive an email from Amazon Future
+              Engineer to claim your benefits. You will also receive occasional
+              emails from Amazon Future Engineer about new opportunities. You
+              always have the choice to adjust your interest settings or
+              unsubscribe.
+            </div>
+            <Button id="continue" onClick={this.submit}>
+              Continue
+            </Button>
+          </form>
+        </div>
+      </div>
+    );
+  }
+}
+
+// This might be better as pure functional component?
+// Just takes handleChange as argument, returns form?
+class ShippingAddressFormGroup extends React.Component {
```

Comment:
also split this one out?

Reply:
I was wondering about best practice on something like this...is it bad practice to put multiple components in the same file, even if the second component is only ever going to be used by the first?

Reply:
Personally I don't mind a few small ones in one file, even more so if they are only used inside of that file.

Reply:
yeah if it's small and simple it's ok to keep it in one file, personally I err on the side of splitting out to avoid files getting long and confusing.

Reply:
Tried a compromise here -- left the shipping address component in the same file, and split out the confirmation page into it's own component. Also moved all three into their own directory just to clean things up a bit. Sound ok?